File: DRZ/Judge.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;


public class Judge {

	static void read(Scanner s, char[] res) {
		String ss = s.next();
		for (int i=0; i<ss.length(); ++i)
			res[i] = ss.charAt(i);
		res[ss.length()] = 0;
	}

	static public void run(DRZ solution) {
		Scanner s = new Scanner(System.in);
		int mode = s.nextInt();

		char[] info = new char[256];
		char[] info2 = new char[256];

		while (true) {
			int cmd = s.nextInt();
			if (0 == cmd) {
				break;
			} else if (1 == cmd) {
				read(s, info);
				//int res = solution.addCitizenInfo(info);
				solution.addCitizenInfo(info);
				int res = 999999999;
				System.out.println(res);
			} else if (2 == cmd) {
				read(s, info);
				int res = solution.delCitizenInfo(info);
				System.out.println(res);
			} else if (3 == cmd) {
				read(s, info);
				read(s, info2);
				solution.modifyCitizen(info, info2);
			} else if (4 == cmd) {
				read(s, info);
				int res = solution.familyMemberCount(info);
				System.out.println(res);
			}
		}
	}
}

File: DRZ/judge.cpp
#include "judge.h"
#include <iostream>
#include <string>
#include <set>

#include <fstream>
#include <chrono>

using namespace std;

namespace {

	char info[MAX_INFO_LEN];
	char info2[MAX_INFO_LEN];

} // namespace

string testCases = "";
string testPrefix = "res/DRZ0";
string testSuffixIn = ".in";
string testSuffixOut = ".out";


void Judge::run(IDRZ *solution) {

    ifstream ins(testPrefix + testSuffixIn);
    ifstream outs(testPrefix + testSuffixOut);

    auto begin = chrono::high_resolution_clock::now();

    int mode; ins >> mode;

	while (true) {
        int cmd; ins >> cmd;
		if (0 == cmd) {
			break;
		} else if (1 == cmd) {
            ins >> info;
			int res = solution->addCitizenInfo(info);

            int expected;
            outs >> expected;

            if (res != expected) {
                cout << res << " " << expected << endl;
            }
		} else if (2 == cmd) {
            ins >> info;
			int res = solution->delCitizenInfo(info);

            int expected;
            outs >> expected;

            if (res != expected) {
                cout << res << " " << expected << endl;
            }
		} else if (3 == cmd) {
            ins >> info;
            ins >> info2;
			solution->modifyCitizen(info, info2);
		} else if (4 == cmd) {
            ins >> info;
			int res = solution->familyMemberCount(info);

            int expected;
            outs >> expected;

            if (res != expected) {
                cout << res << " " << expected << endl;
            }
		}
	}

    auto end = chrono::high_resolution_clock::now();
    cout << chrono::duration_cast<chrono::milliseconds>(end - begin).count() << endl;
}

